core/exit.go

```diff
@@ -34,18 +32,6 @@ func Purge() {
 				} else {
 					opErr = fmt.Errorf("%s; %s", opErr.Error(), err.Error())
 				}
-				// buffer destroy failed; wipe instead
```

Comment:
I removed it because the first thing an allocator does on `Free()` is to wipe the buffer. So there is no added benefit in doing it here again. What do you think?

Reply:
The point of this part is as a last-ditch attempt to wipe the data if the Mprotect call fails inside Free. So we try to make only the inner region mutable if neccessary and then wipe it.

Reply:
Restored behavior in https://github.com/awnumar/memguard/pull/151/commits/97939685232321ccd4759622d8063d913f6f549f. Please note that this is only applicable for the `page` allocator as the `slab` one cannot protect the inner page because it is shared by multiple buffers...